### DISCUSSION TITLE
Handling un-satisfiable setups

### DISCUSSION THREAD

**Discussion by amih77** [google/or-tools#2981] [Routing (and legacy CP) questions]
Hello,

Setup: VRPTW, 10 vehicles, 35 locations. I have a feasible solution
 
I'm currently facing a situation where all heuristics, `meta` or `first_solution` manage to assign only 34, but when I'm forcing the solver to my known solution it finds it under all heuristics.

I want to ask if there's a method to deal with these cases, or a work around.

I got the full assignment from a solution to the same dataset plus an extra location, up to small variations in the time windows - ones that are not changing the satisfiability of the instance, obviously but rather the ability of the solver to find an assignment for all 35 locations.

**Comment by Mizux** [comment]
Did you enable the Guided Local Search ? did you set a time limit ?

**Comment by amih77** [comment]
@Mizux No guided local search, I have a time limit. Is there an example for me to see?

**Comment by Mizux** [comment]
Look at VRP with time limit samples here: https://github.com/google/or-tools/tree/stable/ortools/constraint_solver/samples
IIRC this sample is provided in C++, Python, Java and C#

e.g.
https://github.com/google/or-tools/blob/45770b833997f827d322e929b1ed4781c4e60d44/ortools/constraint_solver/samples/vrp_with_time_limit.py#L95-L104

**Comment by amih77** [comment]
Thanks @Mizux, before your reply, I found an assignment with `TABU_SEARCH` - not explicitly mentioning any first solution strategy. Now, I ran over all pairs of heuristics and `TABU_SEARCH` and `LOCAL_GUIDED_SEARCH` and find nothing....
